### PR TITLE
add IoTDB data bridge documentation

### PR DESCRIPTION
**Pull request by sstrigler**

### CODE REVIEW

**Review by ieQu1 — DISMISSED** [review]

**Comment by LenaLenaPan** [comment]
hi @sstrigler I assume it will be released with 5.0.4? and also need the Chinese version?

**Comment by sstrigler** [comment]
> hi @sstrigler I assume it will be released with 5.0.4? and also need the Chinese version?

Yes, that's for v5.0.4. And yes, a translation needs to be added.

**Comment by sstrigler** [comment]
@LenaLenaPan should I merge it? Now? Or will you?

File: en_US/data-integration/data-bridge-iotdb.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+1. Go to the Dashboard, and click **Data Integration** -> **Data Bridge** from the left navigation menu.
+
+2. Click **Create** on the top right corner of the page.
+
+3. In the **Create Data Bridge** page, click to select **Apache IoTDB**, and then click **Next**.
+
+4. Input a name for the data bridge. The name should be a combination of upper/lower case letters and numbers.
+
+5. Input the connection information:
+   * **Base URL**: Input `http//localhost:18080`, or the actual hostname/IP if the IoTDB server is running remotely.
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-iotdb.md]:
```suggestion
   * **Base URL**: Input `http://localhost:18080`, or the actual hostname/IP if the IoTDB server is running remotely.
```

File: zh_CN/data-integration/data-bridge-iotdb.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+2. 点击页面右上角的**创建**。
+
+3. 在**创建数据桥接**页面, 点击选择 **Apache IoTDB** 作为数据桥接类型, 然后点击 **下一步**。
+
+4. 输入数据桥接名称，要求是大小写英文字母和数字的组合。
+
+5. 输入 Apache IoTDB 服务器链接信息：
+
+   * **基础 URL**: 输入 `http//localhost:18080`；如果服务器在远程运行，须输入实际 IoTDB 服务器地址和端口。
```
Comment (kjellwinblad) [inline zh_CN/data-integration/data-bridge-iotdb.md]:
```suggestion
   * **基础 URL**: 输入 `http://localhost:18080`；如果服务器在远程运行，须输入实际 IoTDB 服务器地址和端口。
```

File: en_US/data-integration/data-bridge-iotdb.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+   * **Username**: Input the IoTDB username; The default value is `root`.
+   * **Password**: Input the IoTDB password; The default value is `root`.
+   
+6. Configure the **Payload Template**: The default value is an empty string, meaning the message payload will be forwarded as JSON-formatted text to IoTDB without modification.
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-iotdb.md]: The Dashboard UI does not seem to have a Payload Template field. Not sure if this is a bug in the UI or the documenetation.

File: en_US/data-integration/data-bridge-iotdb.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+2. Fill in the connection information for the current EMQX instance. 
+
+   - If you run EMQX locally, you can use the default value.
+   - If you have changed EMQX's default configuration. For example, the configuration change on authentication can require you to type in a username and password.
+
+3. Click **Connect** to connect the client to the EMQX instance.
+
+3. Scroll down to the publish area and type in the following:
+   * **Topic**:`t/test`
+   * **Payload**: Make sure to select `JSON` as the payload encoding and then enter:
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-iotdb.md]: The WebSocket client don't have any option to select `JSON` as the payload encoding as far as I'm able to see.

File: en_US/data-integration/data-bridge-iotdb.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+If the data bridge and rule are successfully created, the message should have been published to the specified time series table in the Apache IoTDB server. You can check this using IoTDB's command line interface. If you're using it from docker as above you can connect the server by using this command from your terminal:
+
+```shell
+    $ docker exec -ti iotdb-service /iotdb/sbin/start-cli.sh -h iotdb-service
+```
+
+Then, from within this console, type the following:
+```sql
+IoTDB> select * from root.sg27
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-iotdb.md]: This did not work for me. The message was inserted in `root.t.hej`. My guess is that t.hej is coming from the MQTT topic which was `t/hej`.

**Review by kjellwinblad — COMMENTED** [review]
I did not find any information about the Device ID field in this document. I think that might be worse adding. For example, I don't think it is described anywhere how the Device ID is extracted from the topic when the field is not set in the payload of the message and not in the bridge settings (/ characters are replaced by . and the id is prefixed by "root.").

**Review by kjellwinblad — COMMENTED** [review]
I did not find any information about the Device ID field in this document. I think that might be worse adding. For example, I don't think it is described anywhere how the Device ID is extracted from the topic when the field is not set in the payload of the message and not in the bridge settings (/ characters are replaced by . and the id is prefixed by "root.").

**Review by kjellwinblad — COMMENTED** [review]
I did not find any information about the Device ID field in this document. I think that might be worse adding. For example, I don't think it is described anywhere how the Device ID is extracted from the topic when the field is not set in the payload of the message and not in the bridge settings (/ characters are replaced by . and the id is prefixed by "root.").

File: en_US/data-integration/data-bridge-iotdb.md (hunk 5 above)
Comment (kjellwinblad) [reply en_US/data-integration/data-bridge-iotdb.md]: This is fixed in the following PR https://github.com/emqx/emqx/pull/10770

File: en_US/data-integration/data-bridge-iotdb.md (hunk 3 above)
Comment (Meggielqk) [reply en_US/data-integration/data-bridge-iotdb.md]: I have already informed Shifan Yu about this. It was a bug and I think they are working on it now.

File: en_US/data-integration/data-bridge-iotdb.md (hunk 4 above)
Comment (Meggielqk) [reply en_US/data-integration/data-bridge-iotdb.md]: I changed "select" to "use".

**Comment by sstrigler** [comment]
@Meggielqk needs update of the zn version after I added some more examples: https://github.com/emqx/emqx-docs/pull/1852/commits/56a125bf5938793143f4f559469eaadac2c23b11

**Review by kjellwinblad — DISMISSED** [review]
Looks good to me!

**Review by wivwiv — DISMISSED** [review]

**Review by Meggielqk — DISMISSED** [review]

**Review by wivwiv — APPROVED** [review]

**Pull request merged**